tokio-util/tests/spawn_pinned.rs

```diff
@@ -191,3 +191,39 @@ async fn tasks_are_balanced() {
     // be on separate workers/threads.
     assert_ne!(thread_id1, thread_id2);
 }
+
+#[tokio::test]
+async fn spawn_by_idx() {
+    let pool = task::LocalPoolHandle::new(3);
+
+    let handle1 = pool.spawn_pinned_by_idx(|| async { std::thread::current().id() }, 0);
+    let handle2 = pool.spawn_pinned_by_idx(|| async { std::thread::current().id() }, 1);
+
+    let loads = pool.get_task_loads_for_each_worker();
+    assert_eq!(loads[0], 1);
+    assert_eq!(loads[1], 1);
+    assert_eq!(loads[2], 0);
```

Comment:
This test is fragile. The tasks could complete before you check the load. Consider using a `tokio::sync::Barrier` to prevent them from stopping before checking these counts.